File: Pythonleetcode/Leetcode122.py
class Solution(object):
    def maxProfit(self, prices):
        sum=0
        for i in range(len(prices)-1):
            if prices[i+1]-prices[i]>0:
                sum=sum+ prices[i+1]-prices[i]
        return sum







print(Solution().maxProfit([7,1,5,3,6,4]))

File: CppPractice/Project1/Project1/Leetcode122.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int tem_min = prices[0];
        for (int i = 0; i < prices.size(); i++)
        {
            if (prices[i]>prices[i-1])
            {
                ans += prices[i] - prices[i - 1];
            }
        }
        return ans;
    }
};